source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -371,6 +373,158 @@ namespace
         return prim;
     }
 
+    PvtPrim* readCollection(const CollectionPtr& src, PvtPrim* parent, PvtStage* stage)
+    {
+        const RtToken name(src->getName());
+
+        PvtPrim* prim = stage->createPrim(parent->getPath(), name, RtCollection::typeName());
+        RtCollection collection(prim->hnd());
+        collection.getIncludeGeom().setValueString(src->getIncludeGeom());
+        collection.getExcludeGeom().setValueString(src->getExcludeGeom());
+
+        // createCollectionConnections() handles creating collection inclusion
+        // connections.
+        return prim;
+    }
+
+    void createCollectionConnections(const vector<CollectionPtr>& collectionElements, PvtPrim* parent)
+    {
+        for (const CollectionPtr& colElement : collectionElements)
+        {
+            PvtPrim* parentCollection = findPrimOrThrow(RtToken(colElement->getName()), parent);
+            for (const CollectionPtr& includeCollection : colElement->getIncludeCollections())
+            {
+                PvtPrim* childCollection = findPrimOrThrow(RtToken(includeCollection->getName()), parent);
+                RtCollection rtCollection(parentCollection->hnd());
+                rtCollection.addCollection(childCollection->hnd());
+            }
+        }
+    }
+
+    PvtPrim* readLook(const LookPtr& src, PvtPrim* parent, PvtStage* stage)
+    {
+        const RtToken name(src->getName());
+
+        PvtPrim* prim = stage->createPrim(parent->getPath(), name, RtLook::typeName());
+        RtLook look(prim->hnd());
+
+        // Create material assignments
+        for (const MaterialAssignPtr assign : src->getMaterialAssigns())
+        {
+            PvtPrim* aprim = stage->createPrim(parent->getPath(), RtToken(assign->getName()), RtMaterialAssign::typeName());
+            RtMaterialAssign rassign(aprim->hnd());
+            
+            PvtPrim* collection = findPrimOrThrow(RtToken(assign->getCollectionString()), parent);
+            rassign.getCollection().addTarget(collection->hnd());
+
+            PvtPrim* material = findPrimOrThrow(RtToken(assign->getMaterial()), parent);
+            rassign.getMaterial().addTarget(material->hnd());
+
+            rassign.getExclusive().getValue().asBool() = assign->getExclusive();
+            rassign.getGeom().getValue().asString() = assign->getActiveGeom();
+        }
+
+        // createLookConnections() handles creating look inheritance
```

Comment:
Also a misplaced comment?

Reply:
Similar to the previous one. It's a note that connection part of the work is not done here.